nexus/src/db/datastore.rs

```diff
@@ -2554,32 +2553,6 @@ impl DataStore {
             })
     }
 
-    pub async fn update_available_artifact_fetch(
```

Comment:
The upside of the mess: we can get rid of hand-rolled functions like this one!  (We should be able to get rid of the built-in user and role ones too but I didn't want this change to get much bigger.)